pkgs/development/python-modules/rdkit/default.nix

```
@@ -0,0 +1,139 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, fetchzip
+, cmake
+, boost
+, catch
+, inchi
+, cairo
+, eigen
+, python
+, rapidjson
+, maeparser
+, coordgenlibs
+, numpy
+, stdenv
+, git
+}:
+
+let
+  external = {
+    avalon = fetchzip {
+      url = "http://sourceforge.net/projects/avalontoolkit/files/AvalonToolkit_1.2/AvalonToolkit_1.2.0.source.tar";
+      sha256 = "0nhxfxckb5a5qs0g148f55yarhncqjgjzcvdskkv9rxi2nrs7160";
+      stripRoot = false;
+    };
+    yaehmop = fetchFromGitHub {
+      owner = "greglandrum";
+      repo = "yaehmop";
+      rev = "1b13b52e2738a77715b1bad876e3b4e93f2b5269";
+      sha256 = "1jp7wz8win4mgwxkaz2gnrgsaaqgln04n2lwgfr96isdv1klf62d";
+    };
+    freesasa = fetchFromGitHub
+      {
+        owner = "mittinatten";
+        repo = "freesasa";
+        rev = "2.0.3";
+        sha256 = "0x686zm9fpyg5647fdgxnxgbwav99nc6ymh4bmkr2063yyda4kzc";
+      };
+  };
+in
+buildPythonPackage rec {
+  version = "2020.09.3";
+  pname = "rdkit";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "Release_2020_09_3";
+    sha256 = "1k1wvzcd1yfx4nhz3iq2aaadzdk5w6sfcb4imhvm7pkbzij0nicx";
+  };
+
+  unpackPhase = ''
+    mkdir -p source/External/AvalonTools/avalon source/External/YAeHMOP/yaehmop source/External/FreeSASA/freesasa
+    cp -r ${src}/* source
+    cp -r ${external.avalon}/SourceDistribution/* source/External/AvalonTools/avalon
+    cp -r ${external.yaehmop}/* source/External/YAeHMOP/yaehmop
+    cp -r ${external.freesasa}/* source/External/FreeSASA/freesasa
+
+    find source -type d -exec chmod 755 {} +
+    cp source/External/FreeSASA/freesasa2.c source/External/FreeSASA/freesasa/src
+    ln -s ${rapidjson} source/External/rapidjson-1.1.0
+  '';
+
+  sourceRoot = "source";
+
+  nativeBuildInputs = [
+    cmake
+    git # required by freesasa
+  ];
+
+  buildInputs = [
+    boost
+    catch
+    inchi
+    eigen
+    cairo
+    rapidjson
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  hardeningDisable = [ "format" ]; # required by yaehmop
+  doCheck = false;
+  dontUseSetuptoolsBuild = true;
+  dontUsePipInstall = true;
+  dontUseSetuptoolsCheck = true;
+
+  preConfigure = ''
+    # Don't want this contacting the git remote during the build
+    substituteInPlace External/YAeHMOP/CMakeLists.txt --replace \
+      'GIT_TAG master' \
+      'DOWNLOAD_COMMAND true'
+
+    # Since we can't expand with bash in cmakeFlags
+    cmakeFlags="$cmakeFlags -DPYTHON_INCLUDE_DIR=$(realpath ${python}/include/python*)"
+    cmakeFlags="$cmakeFlags -DPYTHON_NUMPY_INCLUDE_PATH=$(${python}/bin/python -c 'import numpy; print(numpy.get_include())')"
+    cmakeFlags="$cmakeFlags -DFREESASA_DIR=$(realpath External/FreeSASA/freesasa)"
```

```suggestion
    cmakeFlags="$cmakeFlags -DFREESASA_DIR=$src/External/FreeSASA/freesasa"
```

Also please remove the other realpath.

---

Can't use $src since that doesn't contain the external modules, but I pushed a different solution.